eng/jobs/linux-build.yml

```diff
@@ -0,0 +1,178 @@
+parameters:
+  Jobs: []
+
+jobs:
+- ${{ each job in parameters.Jobs }}: 
+  - job: ${{ job.displayName }}
+    displayName: ${{ job.displayName }}
+    pool: ${{ job.pool }}
+    strategy: ${{ job.strategy }}
+    variables: 
+      ${{ insert }}: ${{ job.variables }}
+
+      CommonMSBuildArgs: /flp:v=diag /clp:v=detailed 
+        /p:TargetArchitecture=x64
+        /p:PortableBuild=false
+        /p:ConfigurationGroup=$(_BuildConfiguration)
+        /p:OSGroup=Linux
+        /p:OfficialBuildId=$(OfficialBuildId)
+        /p:StabilizePackageVersion=$(IsStable)
+
+      SourcesDirectory: '$(Build.SourcesDirectory)'
+
+      # Packaging related variables
+      # Debian
+      CommonDebianRunCommand: run --rm --name centos-7-d485f41-20173404063424ubuntu-14.04-debpkg-e5cf912-20175003025046 
+        -v $(SourcesDirectory):/root/coresetup
+        -v $(Build.StagingDirectory)/sharedFrameworkPublish/:/root/sharedFrameworkPublish/
+        -w=/root/coresetup
+        microsoft/dotnet-buildtools-prereqs:ubuntu-14.04-debpkg-e5cf912-20175003025046
+        /root/coresetup/Tools/msbuild.sh
+      DebianPackagingCommand: $(CommonDebianRunCommand)
+        /root/coresetup/src/pkg/packaging/dir.proj
+        /p:UsePrebuiltPortableBinariesForInstallers=true
+        /p:SharedFrameworkPublishDir=/root/sharedFrameworkPublish/
+        $(CommonMSBuildArgs)
+
+      # RPM
+      CommonRpmRunCommand: run --rm --name centos-7-d485f41-20173404063424rhel-7-rpmpkg-c982313-20174116044113
+        -v $(SourcesDirectory):/root/coresetup
+        -v $(Build.StagingDirectory)/sharedFrameworkPublish/:/root/sharedFrameworkPublish/
+        -w=/root/coresetup
+        microsoft/dotnet-buildtools-prereqs:rhel-7-rpmpkg-c982313-20174116044113
+        /root/coresetup/Tools/msbuild.sh
+      RpmPackagingCommand: $(CommonRpmRunCommand)
+        /root/coresetup/src/pkg/packaging/dir.proj
+        /p:UsePrebuiltPortableBinariesForInstallers=true
+        /p:SharedFrameworkPublishDir=/root/sharedFrameworkPublish/
+        $(CommonMSBuildArgs)
+    steps:
+
+    # Build binary and nuget packages
+    - script: $(RunArguments)
+        ./build.sh
+        $(BuildArguments)
+      displayName: Build
+      workingDirectory: '$(SourcesDirectory)'
+
+    # Publish only if internal and not PR 
+    - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+      - script: $(RunArguments)
+          ./Tools/msbuild.sh
+          ./publish/publish.proj
+          $(PublishArguments)
+          $(_CommonPublishArgs)
+        displayName: Publish
+        workingDirectory: '$(SourcesDirectory)'
+        condition: and(succeeded(), eq(variables._BuildConfiguration, 'Release'))
```

Comment:
Other repos are using `_BuildConfig`.  

It seems odd to conditionalize this step in this way via the `_BuildConfiguration` variable.  Couldn't you define "PublishType" on the matrix and set it to "none" or something for the Debug leg and thus let the build step handle not publishing?  ie, move the logic into the build rather than giving your build yaml this control.  That would make adjusting this easier since it would be in one place.

Reply:
Agreed, will do. Just note that earlier publishtype value was set only once, now it is set in each build leg.